### PR TITLE
Migrate URLEncoding using ktor

### PR DESCRIPTION
**Pull request by tobiasheine**
### Description
This migrates the URLEncoding from a custom implementation based on `org.apache.commons.codec.net.URLCodec.encodeUrl(BitSet, byte[])` with the usage of [ktor](https://ktor.io/).

❗️ BLOCKED BY https://github.com/novoda/aws-v4-signer-java/pull/3 ❗️ 

### Considerations
- Migrate URLEncoding for path and query params to a kotlin file using ktor.
- Replace custom StringExtension to create utf8 `ByteArray` with `kotlinx.serialization.toUtf8Bytes` from kotlinx.serialization, which has been added as part of ktor.

### Tests
I had to delete one tests as I couldn't replicate the exact same functionality using ktor:
Encode a request path containing `/ሴ` and escape it to `/%E1%88%B4`.
The other 105 tests completed.

Imho this is really a corner case and I am not sure this is even needed, but I am also open for suggestions how to fix that.

### CODE REVIEW

File: kmp-library/build.gradle [hunk 1]
```diff
@@ -1,6 +1,4 @@
-plugins {
-    id 'kotlin-multiplatform' version '1.3.11'
-}
+apply plugin: 'kotlin-multiplatform'
```
Comment (tobiasheine) [inline kmp-library/build.gradle:1]: I had to change the way we apply the plugin due to some build issues after applying ktor.

File: kmp-library/src/commonMain/kotlin/com/novoda/aws/v4/signer/hash/Sha256.kt [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package com.novoda.aws.v4.signer.hash
 
+import kotlinx.serialization.toUtf8Bytes
```
Comment (tobiasheine) [inline kmp-library/src/commonMain/kotlin/com/novoda/aws/v4/signer/hash/Sha256.kt:3]: This is part of `io.ktor:ktor-client-json`, so we got rid of our custom extension method, see `com/novoda/aws/v4/signer/hash/StringExtension.kt`

**Review by florianmski — APPROVED** [review]
